File: abstraction.java
abstract class Animal{
	public abstract void talk();
}
class Cow extends Animal{
	public void talk() {
		System.out.println("Moo");
	}
}
class Cat extends Animal{
	public void talk() {
		System.out.println("Miyav");
	}
}
class Dog extends Animal{
	public void talk() {
		System.out.println("Hav Hav");
	}
}
class Test{
public static void main(String[] args) {
	Animal a[] = new Animal[3];
	a[0] = new Cat();
	a[1] = new Dog();
	a[2] = new Cow();
	System.out.println("Animals talking");
	for(int i = 0; i < 3; i++)
		a[i].talk();
}

}

File: abstraction.cpp
#include<iostream>
#include<cstring>
using namespace std;
class Animal{
	public :
		virtual void talk() = 0;
};
class Cow : public Animal{
	public : 
	void talk(){
		cout << "Moo" << endl;
	}
};
class Dog : public Animal{
	public : 
	void talk(){
		cout << "Hav Hav" << endl;
	}
};
class Cat : public Animal{
	public : 
	void talk(){
		cout << "Miyav" << endl;
	}
};
int main(){
	Animal *ptr[3];
	ptr[0] = new Cow;
	ptr[1] = new Dog;
	ptr[2] = new Cat;
	cout << "Animals talking" << endl;
	for(int i = 0; i < 3; i++)
		ptr[i] -> talk();
	return 0;
}
